server/src/controllers/base/load/balancedReward.ts

```diff
@@ -1,7 +1,36 @@
 import { KorathReward, Reward } from "../../../enums/Rewards";
-import { Save } from "../../../models/save.model";
+import { FieldData, Save } from "../../../models/save.model";
 import { ORMContext } from "../../../server";
 
+
+/**
+ * Returns the Town Hall building object contained in the given buildingData object.
+ * Throws an error if the town hall cannot be found.
+ * 
+ * The object is identified by checking the `t` (type) variable on the child objects. 
+ * If `t` is equal to 14, then it is considered a town hall.
+ * 
+ * @param buildingData the buildingData object to search
+ * @returns the townHall object found
+ * @throws Error when townHall Object cannot be found
+ */
+function parseTownhallFromBuildingData(buildingData: FieldData) {
+  let townHall: any = null;
+  for(const key in buildingData) {
+    const building = buildingData[key];
+    if(!building) {
+      continue;
+    }
+    if (building.t !== 14) continue;
+
+    townHall = building;
+    break;
```

Comment:
line 21 - 27 can be simplified to
```
if (building && building.t === 14) {
  return building;
}
```
and by doing that line 18 & line 29 can be removed

Reply:
Indeed much cleaner! Thanks!